src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -621,14 +621,16 @@ impl crate::CommandRunner for StreamingCommandRunner {
     let response = with_workunit(
       context.workunit_store.clone(),
       "run_execute_request".to_owned(),
-      WorkunitMetadata::new(),
+      WorkunitMetadata::with_level(Level::Debug),
       timeout_fut,
-      |_, mut metadata| {
-        metadata.level = Level::Error;
-        metadata.desc = Some(format!(
-          "remote execution timed out after {:?}",
-          deadline_duration
-        ));
+      |result, mut metadata| {
+        if let Err(_) = result {
+          metadata.level = Level::Error;
+          metadata.desc = Some(format!(
+            "remote execution timed out after {:?}",
```

Comment:
Should this append the error content to the workunit's existing description?

We definitely need a better way to attach success/failure info than mutating the description. Relates to what @gshuflin is looking at for test results as well.

Reply:
Probably except that this sub-workunit has a blank description ordinarily. Maybe that description should be set to the parent's description always and then append this error message?

Reply:
This particular sub-workunit tracks the actual submission of the execute request so its description probably should either be the same as the parent description or else should be the description on the execute request.

Reply:
We need to figure out our conventions, but: my feeling is that the description for a workunit should be specific to that workunit, rather than its parent.

When we consume a workunit, context is everything, and so we should effectively always be walking the parent link to get more context.